python/IcebergSinkWithKDS/README.md

```diff
@@ -0,0 +1,249 @@
+additional Runtime Properties, as part of the application configuration:
+
+| Group ID                              | Key       | Mandatory | Value                          | Notes                                                                     |
+|---------------------------------------|-----------|-----------|--------------------------------|---------------------------------------------------------------------------|
+| `kinesis.analytics.flink.run.options` | `python`  | Y         | `main.py`                      | The Python script containing the main() method to start the job.          |
+| `kinesis.analytics.flink.run.options` | `jarfile` | Y         | `lib/pyflink-dependencies.jar` | Location (inside the zip) of the fat-jar containing all jar dependencies. |
+
+> ⚠️ If you forget adding these parameters to the Runtime properties, the application will not start.
+---
+
+### How to run and build the application
+
+#### Local development - in the IDE
+
+1. Make sure you have created the Kinesis Streams and you have a valid AWS session that allows you to publish to the Streams (the way of doing it depends on your setup)
+2. Run `mvn package` once, from this directory. This step is required to download the jar dependencies - the Kinesis connector in this case
+3. Set the environment variable `IS_LOCAL=true`. You can do from the prompt or in the run profile of the IDE
+4. Run `main.py`
+
+You can also run the python script directly from the command line, like `python main.py`. This still require running `mvn package` before.
+
+If you are using Virtual Environments, make sure the to select the venv as a runtime in your IDE.
+
+If you forget the set the environment variable `IS_LOCAL=true` or forget to run `mvn package` the application fails on start.
+
+> 🚨 The application does not log or print anything. 
+> If you do not see any output in the console, it does not mean the application is not running.
+> The output is sent to the Kinesis streams. You can inspect the content of the streams using the Data Viewer in the Kinesis console
+
+Note: if you modify the Python code, you do not need to re-run `mvn package` before running the application locally.
+
+##### Troubleshooting the application when running locally
+
+By default, the PyFlink application running locally does not send logs to the console. 
+Any exception thrown by the Flink runtime (i.e. not due to Python error) will not appear in the console. 
+The application may appear to be running, but actually continuously failing and restarting.
+
+To see any error messages, you need to inspect the Flink logs.
+By default, PyFlink will send logs to the directory where the PyFlink module is installed (Flink home).
+Use this command to find the directory:
+
+```
+$ python -c "import pyflink;import os;print(os.path.dirname(os.path.abspath(pyflink.__file__))+'/log')"
+```
+
+
+#### Deploy and run on Amazon Managed Service for Apache Flink
+
+1. Make sure you have the 4 required Kinesis Streams
```

Comment:
Please amend. Only 1 stream (or better none) and the destination S3 bucket is also required